### ISSUE TITLE
Poor metrics for dataset v1.3.0 which features mask, centroid shifts, and z-axis ccd misalignment

### ISSUE THREAD

**Issue by jeipollack** [CosmoStat/wf-psf#171]
Runs of WaveDiff on this dataset show very poor metrics when a realistic mask is applied compared to when it is not.

| Config                                      |        Absolute RMSE +/- STD             | Relative RMSE                       | OPD Metrics ABS RMSE                      | OPD Metrics Relative RMSE(%)                       | SR-GT stars ABS RMSE                   | SR-GT stars Relative RMSE(%)                       |
|---------------------------------------------|----------------------------------|-----------------------------------|----------------------------------|-----------------------------------|----------------------------------|-----------------------------------|
| Case Study (latest) - chkp wgts             | 9.7454e-04 ± 1.1041e-03         | 1.7683e+01 % ± 4.5347e+01 %       | 1.7404e-01 ± 3.1074e-02         | 2.2361e+02 % ± 3.5861e+01 %       | 2.5235e-04 ± 3.5702e-04         | 1.6673e+01 % ± 2.3378e+01 %       |
| Case Study (latest) paper config            | 1.0064e-03 ± 1.0992e-03         | 1.7962e+01 % ± 4.3806e+01 %       | 1.9942e-01 ± 3.0419e-02         | 2.5685e+02 % ± 3.7987e+01 %       | 2.7533e-04 ± 3.5570e-04         | 1.8132e+01 % ± 2.3221e+01 %       |
| gh-165 - chkp wgts paper config             | 1.0064e-03 ± 1.0992e-03         | 1.7962e+01 % ± 4.3806e+01 %       | 1.8775e-01 ± 2.7962e-02         | 1.9584e+02 % ± 3.4340e+01 %       | 2.7533e-04 ± 3.5570e-04         | 1.8132e+01 % ± 2.3221e+01 %       |
| Case Study (latest) different training hyperparams | 1.0734e-03 ± 1.0732e-03   | 1.8701e+01 % ± 4.3202e+01 %       | 2.3238e-01 ± 4.0950e-02         | 2.9771e+02 % ± 3.9742e+01 %       | 2.7877e-04 ± 3.5393e-04         | 1.8351e+01 % ± 2.3120e+01 %       |

This could be a result of the centroid estimation being affected significantly for stars where the mask occludes the cores.

**Comment by jeipollack** [comment]
Here are some plots of the estimated Zernike tip and tilt corrections versus those stored in the dataset during generation as `zernike_centroid_shifts` in both the v1xx training and test datasets.  It was observed to be flipped. Thus, the computed tip and tilt (Z1, Z2) without masks were flipped.

<img width="922" height="355" alt="Image" src="https://github.com/user-attachments/assets/bca4a460-749a-4e19-9cd5-2ca6f9443553" />

here is a scatter plot of the tip and tilt when the mask is applied:

<img width="901" height="359" alt="Image" src="https://github.com/user-attachments/assets/c99adc64-302b-4117-90d3-f9e504a9e1c7" />

We see large scatter and weakening correlation between computed and true shifts with respect to the case when a mask isn't applied. RMSE and Pearson correlation coefficients were computed for each case to show this quantitatively.

Metric | Unmasked | Masked
-- | -- | --
RMSE Total | 0.0145 | 0.0407
RMSE X | 0.0166 | 0.0419
RMSE Y | 0.0119 | 0.0395
Pearson Corr X | 0.9384 | 0.7048
Pearson Corr Y | 0.9720 | 0.7237
Mean Angle Diff (°) | 16.08 | 22.87

I am doing a run passing in the Zernike tip and tilt as a prior and turning off centroid correction for the v1.3.0

**Comment by tobias-liaudat** [comment]
@jeipollack I did not follow what was flipped here

Are the high error stars the ones where the centroid estimation is giving high values (like abs()>0.1)? How do the mask and the star look in those cases?

**Comment by jeipollack** [comment]
I completed my analysis confirming that centroid estimation errors are impacted by masks that occlude star cores, leading to degraded model performance.

My objective was to compare pre-training bad centroids (due to masking artifacts) with post-training poor predictions (identified via high RMSE residuals) to determine whether:

- The model struggles with inherently difficult stars (e.g. due to complexity, location, size),
- Or whether mask-induced centroid errors are the primary cause of performance degradation.

------------

## Centroid Comparison (True vs Masked)
Here is the comparison of "true centroids" (estimated without mask) vs "masked centroids" (estimated with mask) on the test dataset:

<img width="909" height="351" alt="Image" src="https://github.com/user-attachments/assets/9bd37bb5-a7e4-4bb3-95a2-655f6eeff7cf" />

Using the third quartile (Q3) threshold, I removed stars with residuals above the 75th percentile. This produced a tighter correlation:

<img width="923" height="358" alt="Image" src="https://github.com/user-attachments/assets/5ee3622d-5903-4942-a806-5d278d0f6c41" />

**Summary table:**

Metric | Before Outlier Removal | After Q3-Based Outlier Removal
-- | -- | --
RMSE Total | 0.147 | 0.0265
RMSE X | 0.1324 | 0.0266
RMSE Y | 0.1602 | 0.0264
Corr X | 0.9177 | 0.9963
Corr Y | 0.8817 | 0.9964
Mean Angle Diff (°) | 0.1877 | 0.0467

-----------

Here are a couple of images of the mask versus the image with large residuals

<img width="687" height="731" alt="Image" src="https://github.com/user-attachments/assets/0499c756-2a92-417b-b7c2-9879122bc9c1" />

---------

## Training on New dataset ( Q3 Centroid Outliers Removed)

Just to be clear this dataset was generated looking solely at the centroid errors not the model errors post-training.  Using the cleaned dataset (centroid outliers removed), I trained the PSF model using both true centroids (no masks) and masked centroids.

I used this dataset (1500 training stars and 300 test stars) to train the PSF model with case study latest branch (see [spreadsheet](https://docs.google.com/spreadsheets/d/1jIyvXOCuHDY8P548oaWMtbU2hrLgxuKEvgL8D92dNOU/edit?gid=0#gid=0) for full results).  Here's an excerpt of the results:

Training Setup | Abs RMSE (pixels) | Relative RMSE (%) | OPD RMSE | Relative OPD RMSE (%) | SR Abs RMSE | SR Relative RMSE (%)
-- | -- | -- | -- | -- | -- | --
True Centroids (no masks) | 2.07e-04 +/- 1.19e-04 | 2.70 ± 1.88 | 1.39e-01 ± 2.58e-02 | 146.15 ± 25.38 | 5.78e-05 ± 2.69e-05 | 3.79 ± 2.00
Masked Centroids | 1.97e-04 ± 1.07e-04 | 2.47 ± 1.42 | 1.50e-01 ± 2.79e-02 | 193.89 ± 18.59 | 5.11e-05 ± 2.57e-05 | 3.35 ± 1.79
No Outlier Removal | 9.74e-04 +/- 1.10e-03 | 17.68 % +/- 45.35 % | 1.74e-01 +/- 3.101e-02 | 223.61 % +/- 35.861 % | 2.52e-04 +/- 3.57e-04 | 16.67 % +/- 23.38 %

This confirms the benefit of filtering centroid outliers, even when using masked centroids.

----------

## Outlier Analysis

I identified outliers from the poorly trained model using relative RMSE, then compared these indices with those flagged before training based on high centroid residuals.

**Venn diagram of outlier overlap**

<img width="471" height="378" alt="Image" src="https://github.com/user-attachments/assets/8de8df69-5b16-455d-86a2-7b492b14e97c" />

- 75 stars overlap (bad centroids + high residuals)
- 25 only flagged by pre-training centroid error
- 25 only flagged by post-training RMSE

**Diagnostic Scatter plot for the test star sample**

- All test star indices on the x-axis
- Y-axis: RMSE (pixels) per star

**Color Overlays:**
  - Orange: Stars with bad centroids
  - Red (×): Stars with high model residuals

<img width="951" height="387" alt="Image" src="https://github.com/user-attachments/assets/3634a76d-fb97-4f8c-86fd-6fb89457d4de" />

--------

These results should confirm strongly that centroid errors due to masking impact model performance as we suspected and observed with SHE obs ID 2378 analysis presented by Nada, shown in the 9th and 10th rows of her table in [Gitlab Issue 8]( https://gitlab.euclid-sgs.uk/SDC-FR/she_psf_wavediff/-/issues/8)

**Comment by jeipollack** [comment]
> [@jeipollack](https://github.com/jeipollack) I did not follow what was flipped here
> 
> Are the high error stars the ones where the centroid estimation is giving high values (like abs()>0.1)? How do the mask and the star look in those cases?

@tobias-liaudat I updated my reply above. I was speaking of the `zernike_centroid_shifts` as well as `pix_centroid_shifts` in the datasets compared to what is computed by the code. WaveDiff uses a different sign convention than used in the data generation script.

**Comment by jeipollack** [comment]
pinging @tobias-liaudat & for the record, here's the notebook used for this analysis. It is consistent with the version of the code in the `develop` branch.

[Tip_Tilt_Validation_sim_dataset_v1.3.0.ipynb](https://github.com/user-attachments/files/21968197/Tip_Tilt_Validation_sim_dataset_v1.3.0.ipynb)
